homeassistant/components/light/yeelight.py

```diff
@@ -286,13 +328,428 @@ def set_flash(self, flash) -> None:
             except BulbException as ex:
                 _LOGGER.error("Unable to set flash: %s", ex)
 
+    @_cmd
+    def set_effect(self, effect) -> None:
+        """Activate effect."""
+        if effect:
+            from yeelight import (RGBTransition, TemperatureTransition,
+                                  SleepTransition, Flow, BulbException)
+            transition = int(self.config[CONF_TRANSITION])
+            if effect == EFFECT_ALARM:
```

Comment:
Thanks for your response. I think you are right, that the best place for it is in the `yeelight` module. I have tried to make a start with adding it there. But with trial and error i'm getting nowhere. I have no clue on how to start the implementation there.  I will still be trying, but i'm not certain i will get it done :smiley: 
What do you want me to do with this PR?